### PR TITLE
Move Student School to School test scenario

### PR DESCRIPTION
**Pull request by marijanaj**

### CODE REVIEW

File: src/test/java/com/hmhco/hmof/pages/HmofHomePage.java [hunk 1]
```diff
@@ -31,6 +27,27 @@
...
+    @FindBy(id = "caption-0-3")
+    public WebElementFacade button_students;
+
+    @FindBy(id = "caption-0-7")
+    public WebElementFacade button_products;
+
+
+
+    // frame locators
+    @FindBy(css = "frame[name='header']")
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/HmofHomePage.java]: i think Nesib already has code in to switch frames and it looks common. Could'nt you use those?

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 2]
```diff
@@ -1,7 +1,92 @@
...
+
+    @FindBy(name = "newschool")
+    public WebElementFacade dropdownmenu_newSchool;
+
+    @FindBy(xpath = "//select[@name=\"newschool\"]/option[@value=\"01114721\"]")
+    public WebElementFacade dropdownlist_newSchool;
+
+
+    @FindBy(xpath = "//select[@name=\"newschool\"]/option[@value=\"01114719\"]")
+    public WebElementFacade dropdownlist_newSchoolRetrne;
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: typo - Retrne

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 3]
```diff
@@ -1,7 +1,92 @@
...
+
+
+
+    public void selectStudent() throws Throwable {
+        Assert.assertTrue("Checkbox is Visible", hmofHomePage.getElementState(checkbox_student, 5).isCurrentlyVisible());
+        checkbox_student.click();
+        button_next.click();
+    }
+
+    public void backStudentFromNewSchoolInOldSchool() throws Throwable {
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: not sure what this means?

File: src/test/java/com/hmhco/hmof/steps/serenity/StudentsPageSteps.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        moveFromSchoolToSchoolPage.button_done.click();
+    }
+
+    @Step
+    public void baskStudentFromNewSchoolInOldSchool() throws Throwable {
+        moveFromSchoolToSchoolPage.backStudentFromNewSchoolInOldSchool();
+
+    }
+    @Step
+    public void moveFromSchoolToSchool() throws Throwable {
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/steps/serenity/StudentsPageSteps.java]: i guess this should be moveStudentFromSchoolToSchool ?

File: src/test/java/com/hmhco/hmof/steps/serenity/StudentsPageSteps.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        clickLinkMoveFromClassToClass();
+        hmofHomePage.switchToMain();
+        selectOldAndNewSchool();
+        selectStudentForMoving();
+        clickButtonDone();
+    }
+
+
+    @Step
+    public void backStudentInOldSchool() throws Throwable {
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/steps/serenity/StudentsPageSteps.java]: not sure what this means? does it mean something like 'move student back from old school'?

File: src/test/resources/features/RegressionDA/MovingStudents.feature [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Feature: Login to the HMOF Application and DA can Mobility student
+
+  Scenario: To verify DA can move Student School to School
```
Comment (dgracehmh) [inline src/test/resources/features/RegressionDA/MovingStudents.feature]: should this be - To verify DA can move Student from School to School

File: src/test/resources/features/RegressionDA/MovingStudents.feature [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    Given the user is on the HMOF home page
+    When the user logs in with correct credentials 'DATestUser', 'P@ssword1'
```
Comment (dgracehmh) [inline src/test/resources/features/RegressionDA/MovingStudents.feature]: shouldn't there be a Then step here to verify something?

File: src/test/resources/features/RegressionDA/MovingStudents.feature [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+    Given the user open students tab
```
Comment (dgracehmh) [inline src/test/resources/features/RegressionDA/MovingStudents.feature]: This scenario does not make sense. There is no When step and the Then step does not describe what it is verifying

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 9]
```diff
@@ -1,7 +1,92 @@
...
+
+    // locators
+    @FindBy(id = "caption-0-3-3")
+    public WebElementFacade link_moveFromClassToClass;
+
+    @FindBy(name = "oldschool")
+    public WebElementFacade dropdownmenu_orginalSchool;
+
+    @FindBy(xpath = "//select[@name=\"oldschool\"]/option[@value=\"01114719\"]")
+    public WebElementFacade dropdownlist_orginalSchool;
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: are these elements in a drop down list? Are those locators static, i.e. 01114719 ?

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 10]
```diff
@@ -1,7 +1,92 @@
...
+
+
+    @FindBy(xpath = "//select[@name=\"newschool\"]/option[@value=\"01114719\"]")
+    public WebElementFacade dropdownlist_newSchoolRetrne;
+
+
+    @FindBy( xpath="//img[@alt='Next']")
+    public WebElementFacade button_next;
+
+   @FindBy(xpath = "//input[@value=\"f2baeb13a2b54fd0816b339f4a75bd45\"]")
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: is this static - f2baeb13a2b54fd0816b339f4a75bd45 ?

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 11]
```diff
@@ -1,7 +1,92 @@
...
+        public WebElementFacade checkbox_student;
+
+   @FindBy(xpath = "//img[@alt='Done']")
+   public WebElementFacade button_done;
+
+
+
+    public void selectOldAndSelectNewSchool() throws Throwable {
+        Assert.assertTrue("Orginal School drop down menu is Visible", hmofHomePage.getElementState(dropdownmenu_orginalSchool, 5).isCurrentlyVisible());
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: i'm not sure whats going on here? the method seems to be clicking old school twice then clicking new school twice before clicking next. Is this correct?

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 12]
```diff
@@ -1,7 +1,92 @@
...
+        dropdownmenu_newSchool.click();
+        Assert.assertTrue("New School drop down list is Visible", hmofHomePage.getElementState(dropdownlist_newSchool, 5).isCurrentlyVisible());
+        dropdownlist_newSchool.click();
+        Assert.assertTrue("Button next is Visible", hmofHomePage.getElementState(dropdownlist_newSchool, 5).isCurrentlyVisible());
+        button_next.click();
+    }
+
+
+
+    public void selectStudent() throws Throwable {
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: is there a reason we are throwing exceptions in these methods?

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 13]
```diff
@@ -1,7 +1,92 @@
...
+        Assert.assertTrue("Checkbox is Visible", hmofHomePage.getElementState(checkbox_student, 5).isCurrentlyVisible());
+        checkbox_student.click();
+        button_next.click();
+    }
+
+    public void backStudentFromNewSchoolInOldSchool() throws Throwable {
+        Assert.assertTrue("Orginal School drop down menu is Visible", hmofHomePage.getElementState(dropdownmenu_orginalSchool, 5).isCurrentlyVisible());
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: there seems to be lots going on in these methods, assertions, numerous clicks, are all these needed?

File: src/test/java/com/hmhco/hmof/pages/HmofHomePage.java (hunk 1 above)
Comment (marijanaj) [inline src/test/java/com/hmhco/hmof/pages/HmofHomePage.java]: Ok, i will remove.

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java (hunk 2 above)
Comment (marijanaj) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: I will change this locator.

File: src/test/java/com/hmhco/hmof/steps/serenity/StudentsPageSteps.java (hunk 4 above)
Comment (marijanaj) [inline src/test/java/com/hmhco/hmof/steps/serenity/StudentsPageSteps.java]: Yes

File: src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java [hunk 14]
```diff
@@ -1,7 +1,59 @@
...
+import net.serenitybdd.core.pages.WebElementFacade;
+import net.thucydides.core.annotations.DefaultUrl;
+import net.thucydides.core.annotations.Managed;
+import net.thucydides.core.pages.PageObject;
+import org.junit.Assert;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+
+@DefaultUrl("https://my-test.hrw.com/")
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: we dont need this here

Comment (marijanaj) [inline src/test/java/com/hmhco/hmof/pages/Students/MoveFromSchoolToSchoolPage.java]: Ok, i will remove

File: src/test/resources/features/RegressionDA/MovingStudents.feature [hunk 15]
```diff
@@ -0,0 +1,12 @@
+Feature: To verify DA can move Student from School to School
+
+  Scenario: To verify DA can move Student School to School
+    Given the user is on the HMOF home page
+    When the user logs in with correct credentials 'DATestUser', 'P@ssword1'
+
+    Then the user open students tab
```
Comment (dgracehmh) [inline src/test/resources/features/RegressionDA/MovingStudents.feature]: Then steps should be the verification steps. It should be clear from the Step description exactly what is being verified. 'the user open students tab' is not verifying anything. Maybe this Scenario should be structured differently.
Given user is on the HMOF home page
When the user moves the student from an old school to a new school (btw, 'user' here should probably be DA i think?)
Then the student has been moved from old school to new school

That is pretty rough but i hope it shows the idea?

**Pull request closed without merging**